### ISSUE TITLE
add variable validation for essential_contacts_domains_to_allow

### ISSUE THREAD

**Issue by eeaton** [terraform-google-modules/terraform-example-foundation#1261]
### TL;DR

Add [variable validation rules](https://developer.hashicorp.com/terraform/language/values/variables#custom-validation-rules) for variables that are sensitive to unusual/non-intuitive requirements for the accepted format of string.

For example, `essential_contacts_domains_to_allow` is unusual in that it requires an ampersand, "@example.com". A string that doesn't exactly match this format will pass the plan stage but fail the apply stage. Catching these types of mistakes earlier can help reduce friction.

This is particularly disruptive for variables set in the global.tfvars file for the helper script, because correcting the global variables requires deleting everything and starting from a fresh deployment,

### Additional information

anecdotally, I've seen a similar issue with project labels.
